examples/editable-h3-cluster-layer/index.js

```diff
@@ -0,0 +1,194 @@
+        Polygon
+      </Button>
+      <Button
+        onClick={() => setMode(() => DrawCircleFromCenterMode)}
+        active={mode === DrawCircleFromCenterMode}
+      >
+        Circle
+      </Button>
+      <Button onClick={() => setMode(() => DrawRectangleMode)} active={mode === DrawRectangleMode}>
+        Rectangle
+      </Button>
+      <Button onClick={() => setMode(() => DrawPointMode)} active={mode === DrawPointMode}>
+        Point
+      </Button>
+      <div style={{ height: '7px' }} />
+      <Button onClick={() => setBooleanOperation('union')} active={booleanOperation === 'union'}>
+        Union
+      </Button>
+      <Button
+        onClick={() => setBooleanOperation('difference')}
+        active={booleanOperation === 'difference'}
+      >
+        Difference
+      </Button>
+      <Button
+        onClick={() => setBooleanOperation('intersection')}
+        active={booleanOperation === 'intersection'}
+      >
+        Intersection
+      </Button>
+      <div style={{ height: '7px' }} />
+      {renderClusterSelectors(h3Clusters, selectedIndexes, setSelectedIndexes)}
+    </div>
+  );
+}
+
+function renderClusterSelectors(h3Clusters, selectedIndexes, setSelectedIndexes) {
+  return h3Clusters.map((cluster, i) => (
+    <div>
+      <Button
+        key={i}
+        onClick={() => {
+          if (!selectedIndexes.includes(i)) {
+            setSelectedIndexes([...selectedIndexes, i]);
+          } else {
+            setSelectedIndexes(selectedIndexes.filter((index) => index !== i));
+          }
+        }}
+        active={selectedIndexes.includes(i)}
+      >
+        Cluster {i}
+      </Button>
+    </div>
+  ));
+}
+
+export function Example() {
+  const [h3Clusters, setH3Clusters] = React.useState([
+    {
+      hexIds: hexagonCluster1,
+    },
+    {
+      hexIds: hexagonCluster2,
+    },
+    {
+      hexIds: hexagonCluster3,
+    },
+  ]);
+  const [mode, setMode] = React.useState(() => DrawPolygonMode);
+  const [booleanOperation, setBooleanOperation] = React.useState('union');
+  const [selectedIndexes, setSelectedIndexes] = React.useState([0]);
+
+  const layer = new EditableH3ClusterLayer({
+    data: h3Clusters,
+    getHexagons: (x) => x.hexIds,
+    selectedIndexes,
+    resolution: 9,
+    modeConfig: {
+      booleanOperation,
+    },
+    mode,
+    onEdit: ({ updatedData }) => {
+      const newH3Clusters = [...h3Clusters];
+      newH3Clusters[selectedIndexes[0]] = {
+        hexIds: updatedData,
+      };
+      setH3Clusters(newH3Clusters);
+    },
+    _subLayerProps: {
+      'tentative-hexagons': {
+        getFillColor: getTentativeFillColor(booleanOperation),
+      },
+      hexagons: {
+        getFillColor: (cluster) => {
+          if (selectedIndexes.some((i) => h3Clusters[i] === cluster)) {
+            return SELECTED_FILL_COLOR;
+          }
+          return UNSELECTED_FILL_COLOR;
+        },
+        updateTriggers: {
+          getFillColor: selectedIndexes,
```

Comment:
This is the way you can trigger deck.gl to recalculate some prop (in this case `getFillColor`) base on something other than `data`.

Reply:
Ah nice! Great to know.